build.gradle

```diff
@@ -371,94 +491,6 @@ project(":api") {
         withXml = false
     }
 
-    task pom {
```

Comment:
I don't see this task being moved elsewhere. Does this mean the pom.xml is no longer auto-generated?
What was the original reason for auto-generating the pom? One of the benefits of having that was that the developers did not have to remember to update both gradle and maven when they added/updated/removed dependencies, since the release pipeline would run the pom task to generate it. With this gone, we would be going back to release pipeline failures due to missing maven dependencies.

Reply:
Earlier there were multiple POM files. ```pom``` task was there to update these file so we don't have to go & update every file manually. Now I've merged all dependencies into single pom file. If we want to add/update/remove any dependency we just need to update at two places ie. build.gradle & pom.xml. Also sooner or later we are going to deprecate maven all together. So I figured just to remove all maven stuff from build.gradle & keep both build tools isolated from each other.

Reply:
Deprecating maven is a long way out yet, and unfortunately, remembering to change even a single pom file when a corresponding gradle change was made was hard (which was the reason we introduced a special check in the release pipeline to see if either only the build.gradle or pom.xml was changed; and then we removed this special check specifically because introduced the gradle pom command instead) and will happen multiple times guaranteed if we remove this, as it did in the past.

Running a candlepin release only to find out it errored out in the build step, which is way too late in the pipeline, and then causing us to post a PR with the pom.xml fix, and re-running the release from scratch, which then would produce a build by skipping a version because the previous run already tagged a .z version, was very frustrating. We have 2 options here:
1. Re-introduce the check for build.gradle/pom.xml discrepancies early in the release pipeline.
2. Add back the gradlew pom command.
3. Introduce a new jenkins PR pipeline job that checks for build.gradle/pom.xml discrepancies.

I don't like #1, because we still need to create a PR with the pom.xml fix every time someone forgot to do those changes. I'm fine with either #2 or #3, but for #3 we would have to make sure it's introduced before we run a 4.1.z release (if not before this PR is merged).

Reply:
I wonder what others think is the best course of action here: @Januson @wolfdale @sonalidhome @Ceiu

Reply:
@wolfdale You could bring this up as a post-scrum to discuss

Reply:
As discussed, added pom task again.
 
@nikosmoum Since we removed the whole parent-child hierarchy in POM files, we need to make sure every tag (such as profile, repository, scm, license tags in POM file) is generated by ```pom``` task. I still figuring out how to generate [this](https://github.com/candlepin/candlepin/blob/master/pom.xml#L46) via gradle maven plugin.

Reply:
Yes, I think the pom task only used to generate the children poms, not the parent pom.xml

Reply:
Added the gradle pom task. 
We can safely remove the ```activeByDefault``` tag from POM file, since maven profiles are active by default, as we have only one profile defined, we don't have to worry about it.